manager.go

```diff
@@ -182,6 +182,10 @@ func (m *Manager) SubsystemUpdates(ctx context.Context) {
 		m.logger.Error(err)
 	}
 	if needRestart {
+		_, err := InstallNewVersion(ctx, m.logger)
+		if err != nil {
+			m.logger.Errorw("running install of new agent version", "error", err)
+		}
```

Comment:
This accidentally got removed a couple releases ago, so upgrades weren't running the post-install UNLESS "turn_on_hotspot_if_wifi_has_no_internet" was enabled and it was a fresh restart, which caused it to self-update with slightly different logic  right at startup.